Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java

```diff
@@ -31,4 +32,8 @@ public World getWorld() {
     public Location getLocation() {
         return base.getLocation();
     }
+
+    public OfflinePlayer getOffline() {
+        return base.getServer().getOfflinePlayer(base.getUniqueId());
```

Comment:
Does this do a profile lookup if a user isn't in the server's cache? We want to be wary about slow lookups, especially for Bedrock/Vault NPC UUIDs.

Reply:
Mentioned in Discord - but we can potentially skip the lookup by just checking if the base is an instance of `OfflinePlayerStub` and if so, get its base which is itself an OfflinePlayer.

Not sure if this is something we want to do VS just update usages of OfflinePlayer and implement the OfflinePlayer methods that are expected here.

Reply:
Implemented as mentioned above. Should be "good enough" without having to do a lot of extra work to make this play nicer.